src/ray/protobuf/public/events_node_definition_event.proto

```diff
@@ -0,0 +1,32 @@
+// Copyright 2025 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package ray.rpc;
+
+// Message containing the definition of a node, as observed via GCS.
+// The message is expected to be emitted once per node creation.
+//
+// For runtime information associated with this event, see NodeExecutionEvent.
+message NodeDefinitionEvent {
+  bytes node_id = 1;
+  string node_ip_address = 2;
+  // The user-provided identifier or name for this node.
+  string node_name = 3;
+  // Resources (cpu, gpu, etc.) and their counts.
+  map<string, double> resources = 4;
```

Comment:
resources and labels are things that can be changed in the future for example the IPPR work. Better to put in the the NodeExecutionEvent.